### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by myelin0**
## In this branch I implement the following relationships between objects in the app:

-  Belongs-to relationships
-  Has-many relationships
-  Any-to-many relationships
-  Has-many-through relationships

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Classroom
+  attr_accessor :label
+  attr_reader :student
+
```
Comment (DeliceLydia) [inline classroom.rb]: - [ ] I suggest that you make this `attr_reader: students` not `student` and remember to change it anywhere you used it you will have an `@students = [ ]` then you will push each individual `student`

File: student_class.rb [hunk 2]
```diff
@@ -1,11 +1,17 @@
 require './person_class'
+
 class Student < Person
-  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
-    super(name, age, parent_permission)
+  def initialize(age, classroom, name: 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
```
Comment (DeliceLydia) [inline student_class.rb:7]: - [ ] Please add ` attr_reader :classroom`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (DeliceLydia) [inline book.rb]: - [ ] I suggest that you make this `attr_accessor: rentals`  to be able to access it and modify it

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_reader :book, :person
```
Comment (DeliceLydia) [inline rental.rb]: - [ ] Please make `book` and `person` as well `attr_accessor` to be able to access and modify them

File: person_class.rb [hunk 5]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals, :parent_permission
```
Comment (DeliceLydia) [inline person_class.rb]: - [ ] I suggest that you make `rentals` `attr_accessor` as well to be able to access and modify it

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @myelin0, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights are:
- [ ] No linter errors
- [ ] Followed git workflow


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by AshakaE — APPROVED** [review]
Hi @myelin0 🙋‍♂️,

Thanks for making the requested change 👏

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**